django/contrib/auth/checks.py

```diff
@@ -52,7 +53,11 @@ def check_user_model(app_configs=None, **kwargs):
         )
 
     # Check that the username field is unique
-    if not cls._meta.get_field(cls.USERNAME_FIELD).unique:
+    if not cls._meta.get_field(cls.USERNAME_FIELD).unique and not any(
+        isinstance(constraint, models.UniqueConstraint) and
```

Comment:
This is redundant because `total_unique_constraints` returns only `UniqueConstraint`s.

Reply:
Forgot about that check. Thanks!